controllers/apidefinition_controller.go

```diff
@@ -284,6 +247,105 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{RequeueAfter: queueA}, err
 }
 
+func (r *ApiDefinitionReconciler) processClientCertificateReferences(
+	ctx context.Context,
+	env *environmet.Env,
+	log logr.Logger,
+	upstreamRequestStruct *tykv1alpha1.ApiDefinition,
+) {
+	if len(upstreamRequestStruct.Spec.ClientCertificateRefs) != 0 {
+		clientCerts := make([]string, 0)
+
+		for _, secretName := range upstreamRequestStruct.Spec.ClientCertificateRefs {
+			tykCertID, err := r.checkSecretAndUpload(ctx, secretName, upstreamRequestStruct.Namespace, log, env)
+			if err != nil {
```

Comment:
Do we need to check this statement for each iteration? Maybe we can check before the for loop

Reply:
ah, sorry - I chose the wrong line for this comment. 

I meant this statement [here](https://github.com/TykTechnologies/tyk-operator/pull/494/commits/0758be4002b24184c1df56dc405b441f408da893#diff-8c54e4461cf2d12dfafb2336b25204d2861ee5f16312a35dc834adc2500da87bL314-L316)